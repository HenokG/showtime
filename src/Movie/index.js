import Rating from '../Rating'
import React from 'react';

class Movie extends React.Component {
	render() {
		return <><h3>The {this.props.title}</h3><Rating score={this.props.reviewScore}/></>
	}
}

export default Movie;