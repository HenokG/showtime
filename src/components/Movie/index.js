import Rating from '../Rating';
import React from 'react';
import PropTypes from 'prop-types';

class Movie extends React.Component {
  render() {
    const { title, genre, reviewScore } = this.props;
    return (
      <>
        <h3>The {title}</h3>
        {genre && <small>{genre}</small>}
        <Rating score={reviewScore} />
      </>
    );
  }
}

Movie.propTypes = {
  title: PropTypes.string,
  genre: PropTypes.string,
  reviewScore: PropTypes.number,
};

export default Movie;
