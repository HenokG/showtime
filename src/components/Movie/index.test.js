import { shallow } from 'enzyme';
import React from 'react';
import Movie from './';

describe('Testing Movie Component', () => {
  it('should render movie with title, genre and reviewScore', () => {
    const wrapper = shallow(
      <Movie title="Eternal" genre="superhero" reviewScore={4.8} />
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should not render genre', () => {
    const component = shallow(<Movie title="Dr. Strange" reviewScore={10} />);
    expect(component.find('small').exists()).toBe(false);
  });
});
