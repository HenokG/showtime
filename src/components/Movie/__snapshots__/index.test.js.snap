// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Movie Component should render given title, genre and reviewScore 1`] = `
initialize {
  "0": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "data": "The Eternal",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "h3",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Node {
      "attribs": Object {},
      "children": Array [
        Node {
          "data": "superhero",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "small",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": Node {
        "attribs": Object {},
        "children": Array [
          Node {
            "data": "4.8",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "small",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": Node {
          "children": Array [
            [Circular],
            [Circular],
            [Circular],
          ],
          "name": "root",
          "next": null,
          "parent": null,
          "prev": null,
          "type": "root",
        },
        "prev": [Circular],
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "parent": Node {
        "children": Array [
          [Circular],
          [Circular],
          Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "4.8",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "small",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": [Circular],
      "type": "tag",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "parent": Node {
      "children": Array [
        [Circular],
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "superhero",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "small",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "4.8",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "small",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "4.8",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "small",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "superhero",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "small",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "1": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "data": "superhero",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "small",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Node {
      "attribs": Object {},
      "children": Array [
        Node {
          "data": "4.8",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "small",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": Node {
        "children": Array [
          Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "The Eternal",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h3",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": [Circular],
      "type": "tag",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "parent": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "The Eternal",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h3",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        [Circular],
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "4.8",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "small",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": Node {
      "attribs": Object {},
      "children": Array [
        Node {
          "data": "The Eternal",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "h3",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": Node {
        "children": Array [
          [Circular],
          [Circular],
          Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "4.8",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "small",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": null,
      "type": "tag",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "2": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "data": "4.8",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "small",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "The Eternal",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h3",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "superhero",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "small",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "superhero",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "small",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "The Eternal",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h3",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": Node {
      "attribs": Object {},
      "children": Array [
        Node {
          "data": "superhero",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "small",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": Node {
        "children": Array [
          Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "The Eternal",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h3",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": Node {
        "attribs": Object {},
        "children": Array [
          Node {
            "data": "The Eternal",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h3",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": [Circular],
        "parent": Node {
          "children": Array [
            [Circular],
            [Circular],
            [Circular],
          ],
          "name": "root",
          "next": null,
          "parent": null,
          "prev": null,
          "type": "root",
        },
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 3,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
